### DISCUSSION TITLE
Component proposal: Spinner

### DISCUSSION THREAD

**Discussion by Gennady77** [uswds/uswds#5768] [Proposals]
I try to show the process of loading data and didn't find spinners in the component list. They aren't provided in the system?

**Comment by mejiaj** [comment]
How would this component work? File input uses a spinner by adding the `is-loading` class. The SVG is is called [loader.svg](https://github.com/uswds/uswds/blob/develop/src/img/loader.svg)

**Comment by mgifford** [comment]
This is how CMSgov does it:
https://design.cms.gov/components/spinner/
https://github.com/CMSgov/design-system/blob/master/packages/design-system/src/components/Spinner/Spinner.tsx

or from react:
https://react-bootstrap.github.io/components/spinners/

or from Salesforce
https://www.lightningdesignsystem.com/components/spinners/

These look like a good solutions:
https://codeburst.io/how-to-create-a-simple-css-loading-spinner-make-it-accessible-e5c83c2e464c
https://codepen.io/stuartjnelson/pen/xWyLGR

There are others with web components, but we're not set up for that yet.

**Comment by werdnanoslen** [comment]
I'd also suggest expanding this to specifying loading patterns generally:
1. Inline loading (especially short content, table data, etc)
2. Interaction loading (like CMS' buttons above)
3. Full-page loading (spinner and dark background on full page)

I used to use & work on IBM's Carbon DS and really like their page describing everything: https://carbondesignsystem.com/patterns/loading-pattern/

**Comment by brunerae** [comment]
Adding a link to a recently-created Issue that we've closed in favor of this discussion. https://github.com/uswds/uswds/issues/5887

**Comment by szeinedd-nlr** [comment]
any update on this?

**Reply by mejiaj** [reply]
We're currently working on the Glossary and Pre-fill pattern proposals.

This one isn't quite there yet and is missing some key info. There's guidance in [Getting started with USWDS component discussions - What your discussion needs](https://github.com/uswds/uswds/discussions/5764) to help get this moving forward.

**Comment by ryparker** [comment]
#6535 has a related VA spinner example, including usability and accessibility details. Let's compare it with the loading states described here to see what a shared component would need to cover. Both discussions will stay open.
